### PR TITLE
Fix: MCAD and InstaScale CR not deleted when components are Removed

### PR DESCRIPTION
**Pull request by zdtsw**
## Description
we do not Own resources like the ones from CodeFlare but only the standard ones
thus, when set component of codeflare and ray from Managed to Removed, only comfigmap and rolebindings are cleanedup.
this PR is to add the other resources into DSC
fix: https://github.com/opendatahub-io/opendatahub-operator/issues/510

## How Has This Been Tested?

local build image: 
**quay.io/wenzhou/opendatahub-operator-catalog:v2.9.4-10**  @sutaakar if you would like can use this image for verification. 

first enable 2 components
Screenshot from 2023-09-04 17-34-49

and check two CRs
Screenshot from 2023-09-04 17-34-27
Screenshot from 2023-09-04 17-34-06

Screenshot from 2023-09-01 10-04-13

finally Removed components and check again: no instance from codeflare

Screenshot from 2023-09-04 17-44-21

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by zdtsw** [comment]
local run e2e
```
--- PASS: TestOdhOperator (391.90s)
    --- PASS: TestOdhOperator/validate_operator_pod_is_running (76.77s)
        --- PASS: TestOdhOperator/validate_operator_pod_is_running/validate_ODH_Operator_pod (10.12s)
        --- PASS: TestOdhOperator/validate_operator_pod_is_running/validate_CRDs_owned_by_the_operator (62.41s)
    --- PASS: TestOdhOperator/create_Opendatahub_components (237.67s)
        --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test (233.78s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Creation_of_DataScienceCluster_instance (10.68s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components (60.40s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches (10.53s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray (10.70s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines (10.70s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard (10.71s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve (120.40s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing (120.40s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare (120.40s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Ownerrefrences_exist (0.30s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Controller_reconcile (20.85s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field (21.15s)
    --- PASS: TestOdhOperator/delete_components (77.46s)
        --- PASS: TestOdhOperator/delete_components/e2e-test (72.03s)
            --- PASS: TestOdhOperator/delete_components/e2e-test/Deletion:_DataScienceCluster_instance (0.45s)
            --- PASS: TestOdhOperator/delete_components/e2e-test/Deletion:_Application_Resource (71.58s)
PASS
ok      github.com/opendatahub-io/opendatahub-operator/v2/tests/e2e     391.908s
```

**Comment by zdtsw** [comment]
looks like env problem : {"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"k8s.io/test-infra/prow/entrypoint/run.go:84","func":"k8s.io/test-infra/prow/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2023-09-05T07:16:06Z"}

**Comment by zdtsw** [comment]
> @zdtsw Error log
> 
> ```
> === RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field
>     dsc_creation_test.go:47: 
>         	Error Trace:	/go/src/github.com/opendatahub-io/opendatahub-operator/tests/e2e/dsc_creation_test.go:47
>         	Error:      	Received unexpected error:
>         	            	error getting component resource after reconcile: resource name may not be empty
>         	Test:       	TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field
>         	Messages:   	error testing component enabled field
> ```

i will do a local run again. did not see this yesterday.

**Comment by zdtsw** [comment]
tested it again:

[wenzhou@wenzhou opendatahub-operator]$ make e2e-test
go test ./tests/e2e/ -run ^TestOdhOperator -v --operator-namespace=opendatahub-operator-system "--skip-deletion=false" -timeout 15m 
=== RUN   TestOdhOperator
=== RUN   TestOdhOperator/validate_operator_pod_is_running
I0906 16:54:07.541092  512185 request.go:682] Waited for 1.047774063s due to client-side throttling, not priority and fairness, request: GET:https://api.wenzhou-rosa-3.wfu1.p3.openshiftapps.com:443/apis/resolution.tekton.dev/v1beta1?timeout=32s
=== RUN   TestOdhOperator/validate_operator_pod_is_running/validate_ODH_Operator_pod
=== RUN   TestOdhOperator/validate_operator_pod_is_running/validate_CRDs_owned_by_the_operator
=== RUN   TestOdhOperator/create_Opendatahub_components
I0906 16:55:22.187083  512185 request.go:682] Waited for 1.050125959s due to client-side throttling, not priority and fairness, request: GET:https://api.wenzhou-rosa-3.wfu1.p3.openshiftapps.com:443/apis/operator.openshift.io/v1alpha1?timeout=32s
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Creation_of_DataScienceCluster_instance
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Ownerrefrences_exist
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Controller_reconcile
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field
=== RUN   TestOdhOperator/delete_components
I0906 16:59:17.713690  512185 request.go:682] Waited for 1.000152543s due to client-side throttling, not priority and fairness, request: GET:https://api.wenzhou-rosa-3.wfu1.p3.openshiftapps.com:443/apis/resolution.tekton.dev/v1beta1?timeout=32s
=== RUN   TestOdhOperator/delete_components/e2e-test
=== RUN   TestOdhOperator/delete_components/e2e-test/Deletion:_DataScienceCluster_instance
=== RUN   TestOdhOperator/delete_components/e2e-test/Deletion:_Application_Resource
--- PASS: TestOdhOperator (385.99s)
    --- PASS: TestOdhOperator/validate_operator_pod_is_running (75.28s)
        --- PASS: TestOdhOperator/validate_operator_pod_is_running/validate_ODH_Operator_pod (10.12s)
        --- PASS: TestOdhOperator/validate_operator_pod_is_running/validate_CRDs_owned_by_the_operator (61.09s)
    --- PASS: TestOdhOperator/create_Opendatahub_components (235.42s)
        --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test (231.94s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Creation_of_DataScienceCluster_instance (10.41s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components (60.37s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray (10.34s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines (10.34s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches (10.34s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard (10.44s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing (120.01s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare (120.01s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve (120.15s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Ownerrefrences_exist (0.22s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Controller_reconcile (20.34s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field (20.46s)
    --- PASS: TestOdhOperator/delete_components (75.28s)
        --- PASS: TestOdhOperator/delete_components/e2e-test (71.32s)
            --- PASS: TestOdhOperator/delete_components/e2e-test/Deletion:_DataScienceCluster_instance (0.29s)
            --- PASS: TestOdhOperator/delete_components/e2e-test/Deletion:_Application_Resource (71.03s)
PASS
ok      github.com/opendatahub-io/opendatahub-operator/v2/tests/e2e     386.001s

**Comment by VaishnaviHire** [comment]
From the operator [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/opendatahub-io_opendatahub-operator/511/pull-ci-opendatahub-io-opendatahub-operator-main-opendatahub-operator-e2e/1699879083422257152/artifacts/opendatahub-operator-e2e/gather-extra/artifacts/pods/openshift-operators_opendatahub-operator-controller-manager-d4569574c-mfj9q_manager.log) in ci , it seems like an expected error, since codeflare operator is not installed.

File: controllers/datasciencecluster/datasciencecluster_controller.go [hunk 1]
```diff
@@ -371,3 +374,21 @@ func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client
 		return nil
 	}
 }
+
```
Comment (VaishnaviHire) [inline controllers/datasciencecluster/datasciencecluster_controller.go:377]: Should this be removed @zdtsw ?

Comment (zdtsw) [reply controllers/datasciencecluster/datasciencecluster_controller.go:377]: ah, the comments, yes, i will clean it up.

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**